wrapper/src/test/java/integration/refactored/host/TestEnvironment.java

```diff
@@ -0,0 +1,669 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package integration.refactored.host;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import integration.refactored.DatabaseEngine;
+import integration.refactored.DatabaseEngineDeployment;
+import integration.refactored.DriverHelper;
+import integration.refactored.TestDatabaseInfo;
+import integration.refactored.TestEnvironmentFeatures;
+import integration.refactored.TestEnvironmentInfo;
+import integration.refactored.TestEnvironmentRequest;
+import integration.refactored.TestInstanceInfo;
+import integration.refactored.TestProxyDatabaseInfo;
+import integration.util.AuroraTestUtility;
+import integration.util.ContainerHelper;
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.logging.Logger;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.ToxiproxyContainer;
+import org.testcontainers.shaded.org.apache.commons.lang3.NotImplementedException;
+import software.amazon.jdbc.util.StringUtils;
+
+public class TestEnvironment implements AutoCloseable {
+
+  private static final Logger LOGGER = Logger.getLogger(TestEnvironment.class.getName());
+
+  private static final String DATABASE_CONTAINER_NAME_PREFIX = "database-container-";
+  private static final String TEST_CONTAINER_NAME = "test-container";
+  private static final String PROXIED_DOMAIN_NAME_SUFFIX = ".proxied";
+  protected static final int PROXY_CONTROL_PORT = 8474;
+
+  private final TestEnvironmentInfo info =
+      new TestEnvironmentInfo(); // only this info is passed to test container
+
+  // The following variables are local to host portion od test environment. They are not shared to a
+  // test container.
+
+  private int numOfInstances;
+  private boolean reuseAuroraDbCluster;
+  private String auroraClusterName; // "cluster-mysql"
+  private String auroraClusterDomain; // "XYZ.us-west-2.rds.amazonaws.com"
```

Comment:
nit: would auroraDomainSuffix be clearer? auroraClusterDomain has me assuming the cluster ID is included, eg `mycluster.cluster-XYZ.us-west-2.rds.amazonaws.com`